### PR TITLE
taking advantage of function is first-class value

### PR DESCRIPTION
**Pull request by 404pnf**
if f(a) equals g(b) then f equals g.

### CODE REVIEW

**Comment by 404pnf** [comment]
Let me rephrase:

We can always simplify

````js
	var f = function (a) { return g(a) }
````

to

````js
	var f = g
````

**Comment by etiktin** [comment]
:+1:

**Comment by 404pnf** [comment]
Thank you!

**Comment by typicode** [comment]
Closing it since the code has changed quite a bit.
But good point and thank you for the PR.
